### PR TITLE
Move snippets to warm code

### PR DESCRIPTION
**Pull request by gita-omr**
- during code cache disclaim, it's beneficial to move snippets into warm code cache
- controlled by -Xjit:moveSnippetsToWarmCode option

### CODE REVIEW

File: compiler/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -363,6 +363,78 @@ OMR::CodeGenerator::generateCodeFromIL()
...
+
+         bbEnd->setChild(0,NULL);
+         bbEnd->setNumChildren(0);
+         }
+
+      tt->insertAfter(gotoTreeTop);
+      }
+   }
+
 void OMR::CodeGenerator::findLastWarmBlock()
```
Comment (0xdaryl) [inline compiler/codegen/OMRCodeGenerator.cpp:438]: I think the name of this function doesn't (or no longer) reflects what it is actually doing.  And it doesn't return anything as most of our "find" functions do.

Can you rename it to something like `prepareLastWarmBlockForCodeSplitting` ?

Comment (gita-omr) [reply compiler/codegen/OMRCodeGenerator.cpp:438]: > I think the name of this function doesn't (or no longer) reflects what it is actually doing

This function has not changed in any significant way in this PR. It's main effect is still to do this:

`lastWarmBlock->setIsLastWarmBlock(); `

I just took some code out and put into a separate function. The purpose of the remaining code is mostly to do the above (to mark the last warm block).

Comment (gita-omr) [reply compiler/codegen/OMRCodeGenerator.cpp:438]: I think it's important to reflect that before we call this function we don't know which block will be the splitting point. This function is looking for it.

Comment (0xdaryl) [reply compiler/codegen/OMRCodeGenerator.cpp:438]: My point is that someone coming across this API will assume this function will find and return the last warm block in a method.  There is no API documentation (via Doxygen comments) provided for this function, so a developer will rely mostly on the name alone to make a determination on what it does.  "Inserting a goto" is not an operation one would expect if one was simply looking for the last block, as is implied by this function name.

My request is simply to rename it to something that more closely reflects what it does.  If you have an alternate suggestion for how the name could be improved or another means for providing clarity, please propose.

To be fair, we (which probably means "me") should have brought this up when the code was initially contributed many months ago, but didn't.  In the spirit of clarifying the code moving forward I am requesting this change now.

Comment (gita-omr) [reply compiler/codegen/OMRCodeGenerator.cpp:438]: We could just document the method. It's a bit hard to describe everything it does in the name. But I can rename as suggested.

**Comment by gita-omr** [comment]
Addressed last comment. @0xdaryl

**Comment by gita-omr** [comment]
Rebased.

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**